lib/location/location_core.c

```diff
@@ -401,22 +419,71 @@ void location_core_event_cb_pgps_request(const struct gps_pgps_request *request)
 {
 	struct location_event_data pgps_request_event_data;
 
+	LOG_DBG("Request P-GPS data from application: "
+		"prediction_count %d, prediction_period_min %d, gps_day %d, time_of_day %d",
+		request->prediction_count,
+		request->prediction_period_min,
+		request->gps_day,
+		request->gps_time_of_day);
+
 	pgps_request_event_data.id = LOCATION_EVT_GNSS_PREDICTION_REQUEST;
+	pgps_request_event_data.method = LOCATION_METHOD_GNSS;
 	pgps_request_event_data.pgps_request = *request;
 	event_handler(&pgps_request_event_data);
 }
 #endif
 
+#if defined(CONFIG_LOCATION_METHOD_CELLULAR_EXTERNAL)
+void location_core_event_cb_cellular_request(struct lte_lc_cells_info *request)
+{
+	struct location_event_data cell_request_event_data;
+
+	cell_request_event_data.id = LOCATION_EVT_CELLULAR_REQUEST;
+	cell_request_event_data.method = LOCATION_METHOD_CELLULAR;
+	cell_request_event_data.cellular_request = *request;
+	event_handler(&cell_request_event_data);
+}
+
+void location_core_cellular_ready(struct location_data *location, bool undefined)
+{
+	if (k_sem_count_get(&location_core_sem) > 0) {
+		LOG_WRN("Location cellular ready notification called "
+			"but no location request pending");
+		return;
+	}
+
+	LOG_DBG("Location cellular ready notification with %s location, undefined=%d",
+		location == NULL ? "NULL" : "valid",
+		undefined);
+
+	current_event_data.method = LOCATION_METHOD_CELLULAR;
+	if (location == NULL) {
+		if (undefined) {
+			current_event_data.id = LOCATION_EVT_UNDEFINED;
+		} else {
+			current_event_data.id = LOCATION_EVT_ERROR;
+		}
+	} else {
+		current_event_data.id = LOCATION_EVT_LOCATION;
+		current_event_data.location = *location;
+	}
+
+	k_work_submit_to_queue(
+		location_core_work_queue_get(),
+		&location_event_cb_work);
+}
+#endif
+
 static void location_core_event_cb_fn(struct k_work *work)
 {
 	char latitude_str[12];
 	char longitude_str[12];
-	char accuracy_str[12];
+	char temp_str[12];
```

Comment:
**question:** why this rename? This removes clarity on what the string contains.

Reply:
Yep, there was several strings such as altitude, speed, heading which I handled with the same temporary variable. I noticed the change when I removed these but left this change as I'm expecting we may want to log later more data as we add stuff info the location event. But I'll revert the change for now.